src/Controls/src/Core/Items/MarshalingObservableCollection.cs

```diff
@@ -59,7 +59,7 @@ void InternalCollectionChanged(object sender, NotifyCollectionChangedEventArgs a
 				args = new ResetNotifyCollectionChangedEventArgs(items);
 			}
 
-			_dispatcher.Dispatch(() => HandleCollectionChange(args));
+			_dispatcher.DispatchIfRequired(() => HandleCollectionChange(args));
```

Comment:
Does this change behavior slightly? Now it'll conditionally dispatch whereas before it always dispatched?

Reply:
I just renamed the extension method. A bit confusing that before it was Invoke and the extension Dispatch would do the checks first. Now the method is Dispatch and the extension is DispatchIfRequred.

Before and after is checking.